### PR TITLE
exomeseq combine reads from multiple lanes

### PR DESCRIPTION
**Pull request by johnbradley**
Changes `NamedFASTQFilePairType` to `FASTQReadPairType ` which now contains an array of files for each read instead of one file.
Adds new tool `concat-gz-files.cwl` to merge reads array of files together. It uses `ShellCommandRequirement` to do this in one step since the [streamable flag is more of a hint](https://github.com/common-workflow-language/cwltool/issues/644). 
The combined filename is based on the sample name with read index appended.

### CODE REVIEW

File: types/bespin-types.yml [hunk 1]
```diff
@@ -6,9 +6,9 @@ types:
       - name: name
         type: string
       - name: file1
```
Comment (dleehr) [inline types/bespin-types.yml]: `read1_files`

File: tools/concat-gz-files.cwl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+baseCommand: zcat
+inputs:
+  files:
+    type: File[]
+    inputBinding:
+      position: 1
+      shellQuote: False
+  pipe:
```
Comment (dleehr) [inline tools/concat-gz-files.cwl]: With the exception of `files` and `output_filename`, the rest of these inputs should probably be  [CommandLineTool arguments](https://www.commonwl.org/v1.0/CommandLineTool.html#CommandLineTool) instead - _Command line bindings which are not directly associated with input parameters_.

Things like the `|` and `gzip` are part of the command line but not related to any input the user would plug in.

The [CommandLineTool spec](https://www.commonwl.org/v1.0/CommandLineTool.html#Input_binding) describes how the command-line is built up from inputs and arguments. 

Beyond that, any techniques we can use to make tools like this more concise (the equivalent shell script is trivial) are encouraged. I showed this during the meeting yesterday as a progress update, and I think the verbosity (compared to a `zcat *.gz | gzip > files.gz`) is discouraging.

If the tool makes sense as a shell script, one thing we could try is writing the shell script via https://www.commonwl.org/v1.0/CommandLineTool.html#InitialWorkDirRequirement.

File: tools/generate-sample-filenames.cwl [hunk 3]
```diff
@@ -27,6 +28,10 @@ expression: >
     var base = inputs.sample_name
 
     return {
+      combined_reads_output_filenames: [
+        makeFilename(base, 'read1', 'fastq.gz'),
```
Comment (dleehr) [inline tools/generate-sample-filenames.cwl]: Let's use `R1`/`R2` instead of read1/read2

File: tools/concat-gz-files.cwl (hunk 2 above)
Comment (johnbradley) [reply tools/concat-gz-files.cwl]: Switched to arguments but it is still pretty verbose. Will look into an inline shell script.

Comment (johnbradley) [reply tools/concat-gz-files.cwl]: I combined pipe and gzip.

**Pull request merged**